game/scripts/vscripts/Sections/Chapter2/SectionCreeps.ts

```diff
@@ -0,0 +1,337 @@
+                    tg.immediate(() => {
+                        goalKillSniper.complete()
+
+                        if (lastHitTimer) Timers.RemoveTimer(lastHitTimer)
+                    })
+                ])
+            ])
+        ]))
+
+    graph.start(GameRules.Addon.context, () => {
+        print("Completed", sectionName)
+        complete()
+    })
+}
+
+const onStop = () => {
+    print("Stopping", sectionName);
+
+    const hero = getPlayerHero();
+    if (hero) {
+        if (hero.HasModifier(modifier_dk_last_hit_chapter2_creeps.name)) {
+            hero.RemoveModifierByName(modifier_dk_last_hit_chapter2_creeps.name);
+        }
+    }
+
+    if (lastHitTimer) {
+        Timers.RemoveTimer(lastHitTimer)
+        lastHitTimer = undefined;
+    }
+
+    const context = GameRules.Addon.context;
+    const radiantCreeps = context[chapter2SpecificKeys.RadiantCreeps]
+    if (radiantCreeps) {
+        if (Array.isArray(radiantCreeps)) {
+            for (const radiantCreep of radiantCreeps) {
+                if (IsValidEntity(radiantCreep)) {
+                    UTIL_Remove(radiantCreep)
+                }
+            }
+        }
+    }
+
+    const direCreeps = context[chapter2SpecificKeys.DireCreeps]
+    if (direCreeps) {
+        if (Array.isArray(direCreeps)) {
+            for (const direCreep of direCreeps) {
+                if (IsValidEntity(direCreep)) {
+                    UTIL_Remove(direCreep)
+                }
+            }
+        }
+    }
+
+    const sniper = context[chapter2SpecificKeys.sniperEnemyHero]
+    if (sniper) {
+        if (IsValidEntity(sniper)) {
+            UTIL_Remove(sniper);
+        }
+        context[chapter2SpecificKeys.sniperEnemyHero] = undefined;
+    }
+
+    if (graph) {
+        graph.stop(GameRules.Addon.context);
+        graph = undefined;
+    }
+}
+
+export const SectionCreeps = new tut.FunctionalSection(
+    SectionName.Chapter2_Creeps,
+    requiredState,
+    onStart,
+    onStop,
+    chapter2CreepsOrderFilter
+);
+
+export function chapter2CreepsOrderFilter(event: ExecuteOrderFilterEvent): boolean {
+    // Allow all orders that aren't done by the player
+    if (event.issuer_player_id_const != findRealPlayerID()) return true;
+
+    if (!canPlayerIssueOrders) return false;
+
+    return true;
+}
+
+function SendCreepToFight(unit: CDOTA_BaseNPC) {
+    const radiantCreepsSpawnLocation = Vector(-6288, 3280, 128)
+    const direCreepsSpawnLocation = Vector(-5911, 5187, 128)
+
+    let fightPosition = unit.GetTeamNumber() == DotaTeam.GOODGUYS ? direCreepsSpawnLocation : radiantCreepsSpawnLocation;
+
+    ExecuteOrderFromTable({
+        OrderType: UnitOrder.ATTACK_MOVE,
+        Position: fightPosition,
+        UnitIndex: unit.entindex()
+    })
+}
+
+function createLaneCreeps(creepNames: string[], location: Vector, team: DotaTeam, sendCreepToFight: boolean, arrayToPush: CDOTA_BaseNPC[] | undefined): CDOTA_BaseNPC[] {
+    if (!arrayToPush) arrayToPush = [];
```

Comment:
just make this a `const arrayToPush = []` and remove it from the parameters, it's not needed and just confusing.

Reply:
Done, good idea.